lamp/lib/channel-handler.php

```diff
@@ -47,19 +47,26 @@ public static function get_course_channels($course_id)
 
         public static function get_users_channels_in_course($uid, $course_id)
         {
-            $course_channels = Channel::get_course_channels($course_id);
+            global $conn;
 
-            $userchannels = array();
+            $sql = "SELECT * FROM `groupMembership` LEFT JOIN `channels` ON `channels`.`ch_id` = `groupMembership`.`ch_id` WHERE `channels`.`course_id` = '" . $conn->real_escape_string($course_id) . "'";
```

Comment:
We should ensure that we add limit clauses to prevent excessively large database responses.
We can then implement pagination with offsets.